File: School-Management-System/Application/Students/Dtos/StudentDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Application.Students.Dtos
{
    public class StudentDto
    {
        public string? Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string GrandFatherName { get; set; }
        public string FatherName { get; set; }
        public string MotherName { get; set; }
        public int Gender { get; set; }
        public int? Age { get; set; }
        public string ParentContactNumber { get; set; }
        public string? ParentEmail { get; set; }
        public string DobNp { get; set; }
        public string DobEn { get; set; }
        public int ProvinceId { get; set; }
        public int DistrictId { get; set; }
        public int MunicipalityId { get; set; }
        public int WardNo { get; set; }
        public string ContactNumber { get; set; }
        public string ClassRoomId { get; set; }
        public string ClassSectionId { get; set; }
        public string SectionId { get; set; }
        public int? RollNumber { get; set; }
    }
}

File: School-Management-System/WebApi/ClientApp/src/app/home/components/student/shared/models/dtos/student.dto.ts
export interface StudentDto {
  id?: string;
  firstName: string;
  lastName: string;
  grandFatherName: string;
  fatherName: string;
  motherName: string;
  gender: number;
  age: number;
  dobNp: string;
  dobEn: string;
  wardNo: number;
  contactNumber: string;
  parentContactNumber: string;
  parentEmail: string;
  classRoomId: string;
  classSectionId: string;
  sectionId: string;
  provinceId: string;
  districtId: string;
  municipalityId: string;
  rollNumber: string;
  isBusRequired: boolean;
}

export interface StudentStatusDto {
  isActive: boolean;
  remarks?: string;
}
